src/app/shared/services/query.service.ts

```diff
@@ -24,6 +31,7 @@ export class QueryService {
   }
 
   previewActions(userActions: UserActions, queryString: string, pageNum: number, pageSize: number): Promise<object[]> {
+    this.resolveMatchMap(userActions);
```

Comment:
if you have to use this, this should return. input-output function parameters are bad and we should avoid them.